### PR TITLE
Add viewBox to SVG canvas.

### PR DESCRIPTION
**Pull request by bac**
Having a viewBox attribute will allow the SVG to be resized via CSS with scaling, not clipping.

### CODE REVIEW

File: canvas.go [hunk 1]
```diff
@@ -269,28 +269,36 @@ func (c *Canvas) servicesGroup(canvas *svg.SVG) {
...
-	width, height := c.layout()
+// Compute the scale.
+func (c *Canvas) computeScale(width, height int) float32 {
 	scale := float32(1)
 	if height > maxHeight {
 		scale = maxHeight / float32(height)
 	}
 	if float32(width)*scale > maxWidth {
-		scale = maxHeight / float32(height)
+		scale = maxWidth / float32(width)
```
Comment (bac) [inline canvas.go]: This seemed like an error to me.  Drive-by code fix.

I also created the function computeScale so it could be more easily tested.

File: canvas.go [hunk 2]
```diff
@@ -269,28 +269,36 @@ func (c *Canvas) servicesGroup(canvas *svg.SVG) {
...
+	width, height := c.layout()
+	scale := c.computeScale(width, height)
 
 	canvas := svg.New(w)
 	canvas.Start(
 		width,
 		height,
-		`style="font-family:Ubuntu, sans-serif;"`,
+		fmt.Sprintf(`style="font-family:Ubuntu, sans-serif;" viewBox="0 0 %d %d"`,
+			width, height),
```
Comment (bac) [inline canvas.go]: Our target size for display is 600px high. I originally computed the viewBox based on that height but determined the SVG was clipped.  Using a viewBox equal to the actual width and height of the image seems to be the correct thing to do.

**Comment by bac** [comment]
QA: in examples do

go run generatesvg.go openstack.yaml > bundle.svg

You can then view bundle.svg in eog or some other viewer and see it is fully rendered and resizes.

Further you can use test html file like the following to see that it displays and scales properly based on the  height setting:

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <title>Cows</title>
  </head>
<body>
<p>here is the svg</p>
bundle</img>
</body>
</html>

File: canvas.go (hunk 2 above)
Comment (makyo) [inline canvas.go]: Correct. Viewbox is a weird attribute that defines both the size of the view portal and also whether or not the image can be scaled instead of clipped.

**Comment by makyo** [comment]
:+1: QA okay, thanks for this!

File: canvas.go (hunk 1 above)
Comment (mhilton) [inline canvas.go]: That's my fault, sorry.

**Comment by mhilton** [comment]
:+1: LGTM
